src/cart/crt_init.c

```diff
@@ -432,19 +432,21 @@ crt_init_opt(crt_group_id_t grpid, uint32_t flags, crt_init_options_t *opt)
 		prov = crt_gdata.cg_init_prov;
 
 		if (opt && opt->cio_sep_override) {
-			if (opt->cio_use_sep)
+			if (opt->cio_use_sep) {
 				set_sep = true;
-			max_num_ctx = opt->cio_ctx_max_num;
+				max_num_ctx = opt->cio_ctx_max_num;
+			}
 		} else {
 			share_addr = false;
 			ctx_num = 0;
 
 			d_getenv_bool("CRT_CTX_SHARE_ADDR", &share_addr);
-			if (share_addr)
-				set_sep = true;
-
 			d_getenv_int("CRT_CTX_NUM", &ctx_num);
-			max_num_ctx = ctx_num;
+
+			if (share_addr) {
```

Comment:
I don't see where this is set to true.

Reply:
Its on line 443 where its read from envariable

Reply:
ah, sorry, carry on